### PR TITLE
#9 Add .editorconfig file

### LINKED ISSUE #9: Add .editorconfig file
If someone didn't have experience with file EditorConfig it helps maintain consistent coding styles for multiple developers working on the same project across various editors and IDEs.

### PR DESCRIPTION
**Pull request by Nataniel-Antosik**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[{*.kt,*.kts}]
+ij_kotlin_code_style_defaults = KOTLIN_OFFICIAL
+
+max_line_length = 140
```
Comment (bartek977) [inline .editorconfig]: Will it be enough? I think with current screens we could increase to 180

**Review by bartek977 — APPROVED** [review]

File: .editorconfig (hunk 1 above)
Comment (Nataniel-Antosik) [reply .editorconfig]: For me, it could be 180, and if we discuss it in near future and decide it is not a good value we can change it.

Comment (Nataniel-Antosik) [reply .editorconfig]: DONE

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+ij_kotlin_name_count_to_use_star_import = 2147483647
+ij_kotlin_name_count_to_use_star_import_for_members = 2147483647
+
+ij_kotlin_keep_blank_lines_in_declarations = 1
+ij_kotlin_keep_blank_lines_in_code = 1
+ij_kotlin_keep_blank_lines_before_right_brace = 0
+
+ij_kotlin_align_multiline_parameters = false
+ij_continuation_indent_size = 4
```
Comment (MarekMacko) [inline .editorconfig:24]: Please setup `indent_style` and `indent_size` as it's the most basic setting

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+ij_kotlin_import_nested_classes = false
+ij_kotlin_imports_layout = *, ^
```
Comment (MarekMacko) [inline .editorconfig:27]: Please add configuration for `end_of_line` and `insert_final_newline` for all files

**Review by MarekMacko — CHANGES_REQUESTED** [review]

File: .editorconfig (hunk 2 above)
Comment (Nataniel-Antosik) [reply .editorconfig:24]: I added:
```
indent_style = tab
indent_size = 4
```
It's ok?

File: .editorconfig (hunk 3 above)
Comment (Nataniel-Antosik) [reply .editorconfig:27]: I added: 
```
insert_final_newline = true
end_of_line = crlf
```
It's ok?

Comment (MarekMacko) [reply .editorconfig:27]: Note "for all files", not only the kotlin one

File: .editorconfig (hunk 2 above)
Comment (MarekMacko) [reply .editorconfig:24]: If it works, then "yes" :)

File: .editorconfig (hunk 3 above)
Comment (Nataniel-Antosik) [reply .editorconfig:27]: Ok, I changed it

Comment (MarekMacko) [reply .editorconfig:27]: 1. Please check once again, which settings I was referring to, in my first comment and which one you have placed in "[*]" section
2. For `end_of_line` use `lf`

Comment (Nataniel-Antosik) [reply .editorconfig:27]: 1. Yes, my bad I swapped  `indent_style, indent_size` with `end_of_line, insert_final_newline` 
2. I used `lf``

Comment (MarekMacko) [reply .editorconfig:27]: Last one small improvement: put `indent_*` settings on top of the kotlin related settings. Just to group them :)

**Review by MarekMacko — APPROVED** [review]

File: .editorconfig (hunk 1 above)
Comment (MarekMacko) [reply .editorconfig]: @Nataniel-Antosik Please create an issue where everyone should post preferred length, for me 180 is too long :)

Comment (Nataniel-Antosik) [reply .editorconfig]: I created issue for this [Max line length in .editorconfig#41](https://github.com/intive/Picover/issues/41)

**Pull request merged**